ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -29,23 +38,77 @@ internal class CustomerCenterViewModelImpl(
         private const val STOP_FLOW_TIMEOUT = 5_000L
     }
 
-    override val state = flow {
-        try {
-            val customerCenterConfigData = purchases.awaitCustomerCenterConfigData()
-            val purchaseInformation = loadPurchaseInformation()
-            emit(CustomerCenterState.Success(customerCenterConfigData, purchaseInformation))
-        } catch (e: PurchasesException) {
-            emit(CustomerCenterState.Error(e.error))
+    private val _state = MutableStateFlow<CustomerCenterState>(CustomerCenterState.Loading)
+
+    override val state = _state
+        .stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.WhileSubscribed(STOP_FLOW_TIMEOUT),
+            initialValue = CustomerCenterState.Loading,
+        )
+
+    init {
+        viewModelScope.launch {
+            try {
+                val customerCenterConfigData = purchases.awaitCustomerCenterConfigData()
+                val purchaseInformation = loadPurchaseInformation()
```

Comment:
Hmm the idea of doing this loading the way it was before (as part of the `flow`) was to only start loading once a subscriber happens, not during VM creation. This makes testing easier since we can create instances of the VM without actually having to mock the loading of the data. Is there a reason we need to do it here?